Add email notification if satis failed

dubrsl opened the pull request:
This small PR add email notification if satis failed.
It use console mail command. So local MTA should work.

anyt committed:
Added email notifications

dubrsl committed:
Fix error message send command

realshadow commented:
Would you mind changing the config file defaults to null, so the email does not get send in fresh install? Also adding info to readme would be helpful.

In future I would like to add multiple notification providers

dubrsl committed:
Set email and message to null, so the email does not get send in fresh install

dubrsl committed:
Merge remote-tracking branch 'upstream/master' into add_providers_history_size

dubrsl committed:
We use option providers: yes. In this case happen https://github.com/composer/satis/issues/494
The fix - is use option providers-history-size (https://github.com/composer/satis/pull/472), which was added in 1.x branch of satis.
So add support of providers-history-size option in SCP
